contrib/sip_proxy/filters/network/source/metadata.h

```diff
@@ -178,6 +178,14 @@ class MessageMetadata : public Logger::Loggable<Logger::Id::filter> {
 
   std::vector<SipHeader>& listHeader(HeaderType type) { return headers_[type]; }
 
+  auto traContext() {
+    auto context = absl::flat_hash_map<std::string, std::string>{};
+    auto fromHeader = listHeader(HeaderType::From).front().text();
+    context.emplace(std::make_pair("method_type", methodStr[methodType()]));
+    context.emplace(std::make_pair("from_header", fromHeader));
+    return context;
+  }
+
```

Comment:
You're right regarding saving the structure for reuse, now it's done.

Regarding the question about having different contexts content for different API queries, we considered that the content schema would be the same for all requests types, no matter they are updates, creates, retrieves or deletes.

We may convert the returned type as an optional, but it will always exist, so we feel it as unnecessary.